web_scraper/chicago_scraper.py

```diff
@@ -0,0 +1,155 @@
+table_titles = [title.text for title in titles]
+
+table_content = []
+wait = WebDriverWait(driver, 5)
+current = 1
+max = 1
+
+table_rows = wait.until(expected_conditions.presence_of_all_elements_located((By.CSS_SELECTOR,'table#paginatedTable tbody tr')))
+
+for row in table_rows:
+    row_data = [item.text for item in row.find_elements(By.TAG_NAME, 'td')] 
+    table_content.append(row_data)
+
+while current < max: 
+    try :
+        next_bttn = driver.find_element(By.ID, 'paginatedTable_next')
+        next_bttn.find_element(By.TAG_NAME, 'a').click()
+        time.sleep(2)
+        current +=1
+        table_rows = wait.until(expected_conditions.presence_of_all_elements_located((By.CSS_SELECTOR,'table#paginatedTable tbody tr')))
+        for row in table_rows:
+            row_data = [item.text for item in row.find_elements(By.TAG_NAME, 'td')]             
+            table_content.append(row_data)
+    except Exception as e:
+        break
+
+data = pd.DataFrame(np.array(table_content), columns = table_titles)
+data['Tags'] = data['License Type'].apply(lambda x: "General" if "General Contractor" in x 
+                                        else "Electrical" if "Electrical Contractor" in x 
+                                        else "Elevator Mechanic" if "Elevator Mechanic Contractor" in x 
+                                        else "Mason" if "Mason Contractor" in x 
+                                        else "Plumbing" if "Plumbing Contractor" in x 
+                                        else "")
+
+for ind in data.index: #data.index is the label for each row, so for every one index in the dataframe
+    '''check if company_name exists in contractors table'''
+    addr = data['Address'][ind]
+    parts = addr.split(",")
+    if len(parts) == 3:
+        town = parts[1]
+    if len(parts) == 4:
+        town = parts[2]
+
+    license_number = data['License Number'][ind]
+    license_type = data['License Type'][ind]
+    license_expiration = data['License Expiration Date'][ind]
+    insurance_expiration = data['Insurance / Bond Expiration Date'][ind]
+    is_active = data['License Inactive?'][ind]
+
+    if license_expiration is None or license_expiration.strip() == "":
+        license_expiration = None
+    if insurance_expiration is None or insurance_expiration.strip() == "": 
+        insurance_expiration = None
+
+    list_company_names = supabase.table("Contractors").select("id").eq("company_name", data['Name'][ind]).execute()
+    if list_company_names.data: 
+        contractor_ids = supabase.table("Contractors").select("id").eq("company_name", data['Name'][ind]).execute()
+        '''obtain generated id '''
+        contractor_id = contractor_ids.data[0]['id']
+        '''
+        use generated id to update the information with new info that was given on the item, do not use the name of the ocmpany anymore to update
+        '''
+        supabase.table("Contractors").update({
```

Comment:
Use constants for frequently used strings (e.g., 'Contractors', 'Licenses', 'paginatedTable').